app/views/questionnaire.py

```diff
@@ -43,7 +43,7 @@ def check_survey_state():
     values = request.view_args
 
     if not _same_survey(values['eq_id'], values['form_type'], values['collection_id']):
```

Comment:
If we throw and exception we could rename to _check_same_survey and we would not longer need the condition